### ISSUE TITLE
Upon login user home share not being mapped

### ISSUE THREAD

**Issue by ParadingLunatic** [BeyondTrust/pbis-open#107]
Version: 8.5.4.334
OS/Distro: Linux Mint 18.3 Cinnamon
Issue/Impact:  When logging in my home share isn't being mapped to what is configured in RemoteHomeDirTemplate.  The directory exists and the configuration setting CreateHomeDir is set to True.

I've scrubbed the computer name, domain name, etc from the logs and replaced them with generic info.

Checking the syslog I get the following errors twice:
Mar  1 22:27:35 computername kernel: [ 8749.201205] No dialect specified on mount. Default has changed to a more secure dialect, SMB2.1 or later (e.g. SMB3), from CIFS (SMB1). To use the less secure SMB1 dialect to access old servers which do not support SMB3 (or SMB2.1) specify vers=1.0 on mount.
Mar  1 22:27:35 computername kernel: [ 8749.369765] CIFS VFS: protocol revalidation - security settings mismatch
Mar  1 22:27:35 computername kernel: [ 8749.372659] CIFS VFS: session ffff8f3112297c00 has no tcon available for a dfs referral request
Mar  1 22:27:35 computername kernel: [ 8749.375183] CIFS VFS: cifs_mount failed w/return code = -5

After a reboot (recently) I get the following error
Mar  1 22:45:43 computername kernel: [   30.570732] No dialect specified on mount. Default has changed to a more secure dialect, SMB2.1 or later (e.g. SMB3), from CIFS (SMB1). To use the less secure SMB1 dialect to access old servers which do not support SMB3 (or SMB2.1) specify vers=1.0 on mount.
Mar  1 22:45:43 computername kernel: [   30.688773] CIFS VFS: Send error in SessSetup = -126
Mar  1 22:45:43 computername kernel: [   30.688783] CIFS VFS: cifs_mount failed w/return code = -126

 1. systemctl status lwsmd.service
Mar 01 22:27:35 computername lsass[1720]: [lsass] Failed mount of //severname/share on /home/local/DOMAIN/username/NetPersonalShare with data sec=krb5,user=username@DOMAIN.LOCAL,uid=1068500049,gid=1068499457,cruid=106850004
Mar 01 22:27:35 computername lsass[1720]: [lsass] Failed mount of //severname/share on /home/local/DOMAIN/username/NetPersonalShare with data sec=krb5i,user=username@DOMAIN.LOCAL,uid=1068500049,gid=1068499457,cruid=10685000
Mar 01 22:27:35 computername lsass[1720]: [lsass] Failed mount of //severname/share on /home/local/DOMAIN/username/NetPersonalShare, error 1342217968 (errno 40161)
Mar 01 22:27:35 computername lsass[1720]: [lsass] Failed to mount directory for user (username), actual error 40161
Mar 01 22:27:35 computername lsass[1720]: [lsass] Failed to open session for user (name = 'username') -> error = 40161, symbol = LW_ERROR_GENERIC_IO, client pid = 28317

 2. /opt/pbis/bin/lwsm list
lwreg          running (container: 1507)
dcerpc         stopped
eventlog       running (container: 1567)
lsass          running (container: 1720)
lwio           running (container: 1673)
netlogon       running (container: 1628)
rdr            running (io: 1673)
reapsysl       running (container: 1784)
usermonitor    stopped

 3. /opt/pbis/domainjoin-cli query
Name = computername
Domain = DOMAIN.LOCAL
Distinguished Name = CN=COMPUTERNAME,OU=ComputersOU,DC=domain,DC=local

 4. pbis status
LSA Server Status:

Compiled daemon version: 8.5.4.334
Packaged product version: 8.5.334.8
Uptime:        0 days 0 hours 6 minutes 23 seconds

[Authentication provider: lsa-activedirectory-provider]

	Status:        Online
	Mode:          Un-provisioned
	Domain:        domain.local
	Domain SID:    S-1-5-21-2079882332-1687708764-748929250
	Forest:        domain.local
	Site:          Default-First-Site-Name
	Online check interval:  300 seconds
	[Trusted Domains: 1]

	[Domain: LESTO]

		DNS Domain:       domain.local
		Netbios name:     DOMAIN
		Forest name:      domain.local
		Trustee DNS name: 
		Client site name: Default-First-Site-Name
		Domain SID:       S-1-5-21-2079882332-1687708764-748929250
		Domain GUID:      00000000-0000-0000-0000-000000000000
		Trust Flags:      [0x001d]
		                  [0x0001 - In forest]
		                  [0x0004 - Tree root]
		                  [0x0008 - Primary]
		                  [0x0010 - Native]
		Trust type:       Up Level
		Trust Attributes: [0x0000]
		Trust Direction:  Primary Domain
		Trust Mode:       In my forest Trust (MFT)
		Domain flags:     [0x0001]
		                  [0x0001 - Primary]

		[Domain Controller (DC) Information]

			DC Name:              DC1.domain.local
			DC Address:           192.168.1.20
			DC Site:              Default-First-Site-Name
			DC Flags:             [0x0001f3fd]
			DC Is PDC:            yes
			DC is time server:    yes
			DC has writeable DS:  yes
			DC is Global Catalog: yes
			DC is running KDC:    yes

		[Global Catalog (GC) Information]

			GC Name:              BDC.domain.local
			GC Address:           192.168.1.151
			GC Site:              Default-First-Site-Name
			GC Flags:             [0x0001f1fc]
			GC Is PDC:            no
			GC is time server:    yes
			GC has writeable DS:  yes
			GC is running KDC:    yes

 5. /opt/pbis/bin/enum-users
There are more but this gives you an idea
User info (Level-0):
====================
Name:              username
Uid:               1068500049
Gid:               1068499457
Gecos:             User Name
Shell:             /bin/bash
Home dir:          /home/local/DOMAIN/username

 6. attach logs
 
[lsass.log](https://github.com/BeyondTrust/pbis-open/files/1773671/lsass.log)

**Comment by rbest-bt** [comment]
Can you list the output of:
> /opt/pbis/bin/config --show RemoteHomeDirTemplate

**Comment by ParadingLunatic** [comment]
The output is %H/local/%D/%U/NetPersonalShare

Also the folder does get created if I delete the folder, log off, and then log back on.

**Comment by rbest-bt** [comment]
Did it also mount? 

We have some issues around the mounting in this policy (also see #10 ).

**Comment by ParadingLunatic** [comment]
It didn't mount at all.  It just creates the folder but it errors trying to mount it.  Hence the errors in the syslog.

**Comment by ParadingLunatic** [comment]
I think PART of my problems were that I didn't have the some of the kerberos utils installed.  I attempted to run some of the commands listed in the other issue you linked.  I didn't have kinit or kvno installed.  After installing them I went through some of the issues and ran the smbclient -k //server.domain.local/share and still couldn't access.  After searching the error it had to do with the SMB version on the server.  The server is OpenMediaVault 3 which runs Debian 8 (jessie).  Set protocol = SMB3 on the server and restarted smb.  After that I was able to access the share successfully via smbclient -k //server.domain.local/share.  Rebooted my laptop logged in, checked the folder.  Still nothing mapped.  Checked syslog and it looks a LITTLE more promising.  As you'll see below, it's still failing trying to mount.  I can still access the share with smbclient -k //server.domain.local/share.  Once again, the info in the syslog has been cleaned but everything it references is correct.  I would imagine the service ticket errors and client principle name errors would probably be the likely culprit.

Mar  2 16:06:32 computername cifs.upcall: key description: cifs.spnego;0;0;39010000;ver=0x2;host=server.domain.local;ip4=192.168.1.5;sec=krb5;uid=0x3fb00451;creduid=0x3fb00451;user=username@DOMAIN.LOCAL;pid=0x747
Mar  2 16:06:32 computername cifs.upcall: ver=2
Mar  2 16:06:32 computername cifs.upcall: host=server.domain.local
Mar  2 16:06:32 computername cifs.upcall: ip=192.168.1.5
Mar  2 16:06:32 computername cifs.upcall: sec=1
Mar  2 16:06:32 computername cifs.upcall: uid=1068500049
Mar  2 16:06:32 computername cifs.upcall: creduid=1068500049
Mar  2 16:06:32 computername cifs.upcall: user=username@DOMAIN.LOCAL
Mar  2 16:06:32 computername cifs.upcall: pid=1863
Mar  2 16:06:32 computername cifs.upcall: find_krb5_cc: scandir error on directory '/run/user/1068500049': No such file or directory
Mar  2 16:06:32 computername cifs.upcall: krb5_get_init_creds_keytab: 13
Mar  2 16:06:32 computername cifs.upcall: handle_krb5_mech: getting service ticket for server.domain.local
Mar  2 16:06:32 computername cifs.upcall: cifs_krb5_get_req: unable to get client principal name
Mar  2 16:06:32 computername cifs.upcall: handle_krb5_mech: failed to obtain service ticket (-1765328189)
Mar  2 16:06:32 computername cifs.upcall: Unable to obtain service ticket
Mar  2 16:06:32 computername cifs.upcall: Exit status -1765328189
Mar  2 16:06:32 computername kernel: [   27.644233] CIFS VFS: cifs_mount failed w/return code = -126
Mar  2 16:06:32 computername kernel: [   27.739865] CIFS VFS: cifs_mount failed w/return code = -126
Mar  2 16:06:32 computername kernel: [   27.891143] CIFS VFS: cifs_mount failed w/return code = -126
Mar  2 16:06:32 computername kernel: [   27.988792] CIFS VFS: cifs_mount failed w/return code = -126

**Comment by ParadingLunatic** [comment]
I think part of my problem is my OpenMediaVault server doesn't have kerberos configured properly.  I just checked the krb5.conf file on there and it's all sorts of wrong.  PART of it is correct but there's a lot that's not right.

**Comment by ParadingLunatic** [comment]
I have a feeling that very well may be my problem.  I created a share on my Server 2016 domain controller and repointed my home directory to that share.  Obviously windows then created a folder in the share.  Switched to a different console and logged in.  Tada, it worked.  It mounted (although I did get an ioctl error but I have a feeling it's not major) the share and I was able to access the test files I put in there, as well as copy/delete/etc as expected.  So I guess now my new task is to fix kerberos on my OMV server.

**Comment by ParadingLunatic** [comment]
So, in the long run, I'm pretty positive the configuration on my OpenMediaVault server is my problem.  The problem is, it's technically working how I want it.  All of my windows systems auto map the home drives fine, just my linux laptop is having the issue.  I've decided I'm not going to spend how ever many more hours trying to figure out how to possibly fix this issue and maybe break everything else.  Instead I created a credential file in my linux home directory and I'm using fstab to mount the share.  It's a lot less time and a lot less headaches.
